front/ui/ui-charts/src/speedSpaceChart/components/SpeedSpaceChart.tsx

```diff
@@ -159,13 +173,13 @@ const SpeedSpaceChart = ({
           style={{ width: adjustedWidthRightAxis }}
         >
           <SettingsPanel
-            globalHeight={dynamicHeight}
             color={backgroundColor}
             store={store}
             setStore={setStore}
             setIsMouseHoveringSettingsPanel={setIsMouseHoveringSettingsPanel}
             translations={translations}
             testIdPrefix="settings-panel"
+            setChartHeight={setHeight}
```

Comment:
What about creating a specific function in `SpeedSpaceChart` to handle height changes? We could pass this function to `SettingsPanel` instead of passing `setChartHeight`.


```typescript
const adjustHeightForLayerChange = useCallback((layerName: string, isCurrentlyActive: boolean) => {
  let adjustment = 0;
  
  switch (layerName) {
    case 'electricalProfiles':
      adjustment = LINEAR_LAYERS_HEIGHTS.ELECTRICAL_PROFILES_HEIGHT;
      break;
    case 'powerRestrictions':
      adjustment = LINEAR_LAYERS_HEIGHTS.POWER_RESTRICTIONS_HEIGHT;
      break;
    case 'speedLimitTags':
      adjustment = LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT;
      break;
    default:
      adjustment = 0;
  }
  
  setHeight(prevHeight => prevHeight + (isCurrentlyActive ? -adjustment : adjustment));
}, [setHeight]);
```

This would make the code easier to understand and maintain because all height calculations would be in one place.

```suggestion
            adjustHeightForLayerChange={adjustHeightForLayerChange}
```

Reply:
Done